internal/querycoord/cluster.go

```diff
@@ -347,6 +349,27 @@ func (c *queryNodeCluster) getSegmentInfo(ctx context.Context, in *querypb.GetSe
 	return segmentInfos, nil
 }
 
+type queryNodeGetMetricsResponse struct {
+	resp *milvuspb.GetMetricsResponse
+	err  error
+}
+
+func (c *queryNodeCluster) getMetrics(ctx context.Context, in *milvuspb.GetMetricsRequest) []queryNodeGetMetricsResponse {
+	c.RLock()
+	defer c.RUnlock()
+
+	ret := make([]queryNodeGetMetricsResponse, 0, len(c.nodes))
+	for _, node := range c.nodes {
+		resp, err := node.getMetrics(ctx, in)
+		ret = append(ret, queryNodeGetMetricsResponse{
```

Comment:
why not return errror directly instead of creating a struct called queryNodeGetMetricsResponse?

Reply:
Returning an error directyly doesn't let us know which response is valid or invalid. And in fact, I think we should only collect the normal response.

Reply:
```go
		if nodeMetrics.err != nil {
			log.Warn("skip invalid metrics of node",
				zap.Error(nodeMetrics.err))
			continue
```
the error info seems can't be return to sdk, so  the error response is aborted discarded.

Reply:
I think it doesn't matter. The client should only care the healthy query nodes.